stream-chat-swift-docs/docs/guides/working-with-members.md

```diff
@@ -1,3 +1,223 @@
+        filter: .equal("invite", to: "pending")
+    ) 
+) 
+```
+
+## Pagination and ordering
+
+Since there potentially can be large amount of members in a channel, querying members supports pagination.
+By default members are ordered from oldest to newest and can be paginated using offset-based pagination:
+
+```swift
+memberListController.loadNextMembers(limit: 10) { error in 
+    // handle error / access members 
+    print(error ?? memberListController.members) 
+}
+```
+
+## Adding/removing members
+
+```swift
+let controller = chatClient.channelController(for: .init(type: .messaging, id: "general")) 
+controller.addMembers(userIds: ["lando", "ann"]) 
+controller.removeMembers(userIds: ["raya"])
+```
+
+:::caution
+You can only add/remove up to 100 members at once.
+:::
+
+### Leaving a channel
+
+It is possible for a user to leave the channel without moderator-level permissions if channel members have `RemoveOwnChannelMembership` permission:
+
+```swift
+let controller = chatClient.channelController(
+    for: .init(type: .messaging, id: "general")
+) 
+controller.removeMembers(userIds: ["my-user-id"])
+```
+
+## Observing changes to members
+
+Set the delegate of `ChatChannelMemberListController` to observe the changes in the system.
+
+The delegate can be set directly only if you're **not** using custom extra data types. Due to the current limits of Swift and the way it handles protocols with associated types, it's required to use `setDelegate` method instead to set the delegate, if you're using custom extra data types.
+
+### Without extra data
+```swift
+class Controller: ChatChannelMemberListControllerDelegate {
+    let memberListController: ChatChannelMemberListController
+
+    init() {
+        memberListController = chatClient.memberListController( 
+            query: .init(cid: .init(type: .messaging, id: "general"), filter: .none) 
+        )
+        memberListController.delegate = self
+    }
+
+    func memberListController(
+        _ controller: ChatChannelMemberListController,
+        didChangeMembers changes: [ListChange<ChatChannelMember>]
+    ) {
+        // process changes
+    }
+}
+```
+
+### With extra data
+```swift
+final class CustomExtraData: ExtraDataTypes { }
+
+final class Controller: _ChatChannelMemberListControllerDelegate {
+    typealias ExtraData = CustomExtraData
+    let memberListController: ChatChannelMemberListController
+
+    init() {
+        memberListController = chatClient.memberListController( 
+            query: .init(cid: .init(type: .messaging, id: "general"), filter: .none) 
+        )
+        memberListController.setDelegate(self)
+    }
+    
+    func memberListController(
+        _ controller: _ChatChannelMemberListController<ExtraData>,
+        didChangeMembers changes: [ListChange<_ChatChannelMember<ExtraData.User>>]
+    ) {
+        // process changes
+    }
+}
+```
+
+// TODO ??
+More information on extra data topic can be found here: ---
```

Comment:
@b-onc  Are we going to add a separate page with explanations on extra data?

Reply:
exactly, in the future